pkg/causality/internal/slots.go

```diff
@@ -82,29 +74,19 @@ func (s *Slots) Add(elem *Node) {
 		// Add this node to the slot, make sure new coming nodes with the same hash should
 		// depend on this node.
 		s.slots[slotIdx].nodes[hash] = elem
+
+		s.slots[slotIdx].mu.Unlock()
 	}
 
 	// Construct the dependency graph based on collected `dependencyNodes` and with corresponding
 	// slots locked.
 	elem.dependOn(dependencyNodes)
```

Comment:
Calling `dependOn` without locks on the slots is dangerous. Because the `Node` it depends may be removed at the same time. The same dependency may be removed both in `elem.dependOn` and `Slots.Remove`.

Reply:
The slot lock just protects queue leaders in the slot. This has been already changed at line 76.

Reply:
ok